packages/docs-reanimated/static/examples/FABButton.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { StyleSheet, SafeAreaView, View, Pressable, Text } from 'react-native';
+import Animated, {
+  withDelay,
+  interpolate,
+  useAnimatedStyle,
+  useSharedValue,
+  withSpring,
+  withTiming,
+} from 'react-native-reanimated';
+
+const AnimatedPressable = Animated.createAnimatedComponent(Pressable);
+
+const SPRING_CONFIG = {
+  duration: 1200,
+  overshootClamping: true,
+  dampingRatio: 0.8,
+};
+
+const OFFSET = 60;
+
+const FABButton = ({ animatedStyles, buttonLetter }) => (
+  <AnimatedPressable style={[animatedStyles, styles.shadow, styles.button]}>
+    <Animated.Text style={styles.content}>{buttonLetter}</Animated.Text>
+  </AnimatedPressable>
+);
+
+export default function App() {
+  const isExpanded = useSharedValue(false);
+
+  const handlePress = () => {
+    isExpanded.value = !isExpanded.value;
+  };
+
+  const generateButtonAnimatedStyle = (index) =>
+    useAnimatedStyle(() => {
+      const moveValue = interpolate(
+        Number(isExpanded.value),
+        [0, 1],
+        //highlight-next-line
+        [0, OFFSET * index]
+      );
+      const translateValue = withSpring(-moveValue, SPRING_CONFIG);
+      //highlight-next-line
+      const delay = index * 100;
+
+      const scaleValue = isExpanded.value ? 1 : 0;
+
+      return {
+        transform: [
+          { translateY: translateValue },
+          {
+            scale: withDelay(delay, withTiming(scaleValue)),
+          },
+        ],
+      };
+    });
+
```

Comment:
I'm not a fan of generating animated styles as it breaks my "rules of hooks" mental model.

I believe it would be cleaner if we had this animated style defined once inside the FloatingActionButton component and pass only the required values (for example `isExpanded` shared value) as props.